man/man1/pkcstok_admin.1.in

```diff
@@ -0,0 +1,192 @@
+.TH pkcstok_admin 1 "July 2024" "@PACKAGE_VERSION@" "openCryptoki"
+.SH NAME
+pkcstok_admin \- utility to administrate token directories of the Opencryptoki
+token repository.
+.
+.
+.
+.SH SYNOPSIS
+.
+.B pkcstok_admin
+.I command
+.RB [ OPTIONS ]
+.PP
+.
+.B pkcstok_admin
+.BR \-\-help | \-h
+.br
+.B pkcstok_admin
+.BR \-\-version | \-v
+.
+.
+.
+.SH DESCRIPTION
+.
+The \fBpkcstok_admin\fP utility can be used to administrate token directories of
+the Opencryptoki token repository with proper permissions and owners.
+By default token directories are owned by the \fB@pkcs_group@\fP group, and
+every user that is a member of the \fB@pkcs_group@\fP group has access to the
+token and its token objects (i.e. keys, certificates, etc.).
+.PP
+A token directory can alternatively be owned by a token specific group.
+Specify the group name with keyword \fBusergroup\fP in the slot definition in
+the \fBopencryptoki.conf\fP configuration file locate in
+\fB@sysconfdir@/opencryptoki/\fP.
+With that, only users that are members of the configured token specific group
+have access to the token and its token objects. Users that are only members of
+the \fB@pkcs_group@\fP group, but are not also a members of the token specific
+group do not have access to the token. Users of the token specific group must
+still also be members of the \fB@pkcs_group@\fP group to be able to use
+Opencryptoki.
+.PP
+Token specific files are usually stored in
+\fB@localstatedir@/lib/opencryptoki/<token-name>/\fP. Additionally token
+specific lock files are stored in
+\fB@localstatedir@/lock/opencryptoki/<token-name>/\fP.
+Furthermore, a token specific POSIX shared memory segment exists under
+\fB/dev/shm/\fP. All files and directories within these token specific
+directories as well as the shared memory segment are owned by the token
+specific group, or by the \fB@pkcs_group@\fP group if no token specific group
+has been configured.
+.PP
+The \fBpkcstok_admin\fP utility must be run as \fBroot\fP, and the
+\fBpkcsslotd\fP must be stopped before running this utility.
+.PP
+.PP
+.B Note:
+Do not use the \fBpkcstok_admin\fP utility on a \fBTPM token\fP. The
```

Comment:
Is this checked in the code? (Not using the tool on a TPM token)

Reply:
No, that's not possible to check. The TPM token can have an arbitrary name (like any other tokens, too), so we can't identify the token type by just looking at the token directory